api/tests.py

```diff
@@ -0,0 +1,193 @@
+from .test_helpers import SmallboardTestCase
+from rest_framework import status
+from puzzles.models import Puzzle, PuzzleTag
+
+if "unittest.util" in __import__("sys").modules:
+    # Show full diff in self.assertEqual.
+    __import__("sys").modules["unittest.util"]._MAX_LENGTH = 999999999
+
+TEST_URL = "https://smallboard.test/"
+
+
+class ApiTests(SmallboardTestCase):
+    def test_get_hunt(self):
+        response = self.get_hunt()
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            response.data,
+            {
+                "id": self._hunt.pk,
+                "name": self._hunt.name,
+                "url": self._hunt.url,
+                "active": self._hunt.active,
+            },
+        )
+
+    def test_create_invalid_puzzle(self):
+        # Missing name
+        response = self.create_puzzle(
+            {"url": "https://smallboard.test", "is_meta": False}
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        # Missing url
+        response = self.create_puzzle({"name": "test name", "is_meta": False})
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        # Empty string name
+        response = self.create_puzzle({"name": "", "url": TEST_URL, "is_meta": False})
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        # Malformatted URL (TODO)
+        # response = self.create_puzzle(
+        #     {"name": "test name", "url": "bad_url", "is_meta": False}
+        # )
+        # self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        self.assertEqual(Puzzle.objects.count(), 0)
+
+    def test_create_puzzle(self):
+        response = self.create_puzzle({"name": "test name", "url": TEST_URL})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        puzzle = Puzzle.objects.get()
+        self.assertEqual(
+            response.data,
+            {
+                "id": puzzle.pk,
+                "name": puzzle.name,
+                "hunt_id": self._hunt.pk,
+                "url": TEST_URL,
+                "notes": "",
+                "sheet": None,
+                "chat_room": {},
+                "status": "SOLVING",
+                "tags": [],
+                "guesses": [],
+                "metas": [],
+                "feeders": [],
+                "is_meta": False,
+            },
+        )
+
+    def test_delete_puzzle(self):
+        self.create_puzzle({"name": "test name", "url": TEST_URL})
+        puzzle = Puzzle.objects.get()
+        response = self.delete_puzzle(puzzle.pk)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, {})
+        self.assertEqual(Puzzle.objects.count(), 0)
+
+    def test_edit_puzzle(self):
+        self.create_puzzle({"name": "test name", "url": TEST_URL})
+        puzzle = Puzzle.objects.get()
+
```

Comment:
Edge cases: 
- Editing more than 1 field at a time
- Editing a non-editable field
- New name matches a puzzle already in DB